CRM/Contribute/Form/ManagePremiums.php

```diff
@@ -55,11 +56,39 @@ public function setDefaultValues() {
       if (isset($tempDefaults['period_type'])) {
         $this->assign('showSubscriptions', TRUE);
       }
+
+      // Convert api3 field names to custom_xx format
+      foreach ($tempDefaults as $name => $value) {
+        if ($new = CRM_Core_BAO_CustomField::getShortNameFromLongName($name)) {
```

Comment:
best not to use assignments in conditionals.
Also the getShortNameFromLongName doesn't guard against non-dot-notation names, so this should probably be 
```suggestion
        if (str_contains($name, '.')) {
        $new = CRM_Core_BAO_CustomField::getShortNameFromLongName($name)
```

Reply:
I had added a guard in the `getShortNameFromLongName`  function, because it was low-cost, and avoids copy-pasta.

https://github.com/civicrm/civicrm-core/pull/31250/files#diff-c3c751ad9f31137a69214a8c0dcedd2fc3788f00431497be6c087d9fe5ca2687R617

I updated to move the assignment outside the conditional.